webapp/src/client/app/shared/services/web-form.service.ts

```diff
@@ -73,4 +86,16 @@ export class WebFormService extends BaseComponent {
   detectFormChanges(cdr: ChangeDetectorRef): void {
     cdr.detectChanges();
   }
+
+  public canConnect(form: FormGroup): boolean {
+    return form.valid && this.canConnectExtraCallback();
+  }
+
+  canConnectIfAlsoTrue(callback: () => boolean): void {
+    this.canConnectExtraCallback = () => callback();
+  }
+
+  resetCanConnectCallback(){
```

Comment:
I'll be picky here and ask to add a space after the parathesis 🧐 Everything else looks great!
`resetCanConnectCallback() {`